core/src/modules/ops/destination.ts

```diff
@@ -912,82 +923,109 @@ export namespace DestinationOps {
     success: boolean;
     error: Error;
   }> {
-    const _exports: Export[] = []; // only ones we are sending
-    for (const _export of givenExports) {
-      // TODO: maybe should recalcuate hasChanges and from current mostRecent
-      if (!_export.hasChanges) {
-        await _export.complete(); // do not include exports with hasChanges=false
-      } else {
-        _exports.push(_export);
-      }
-    }
+    const _exports: Export[] = [];
+    const locks: Awaited<ReturnType<typeof getLock>>[] = [];
 
-    const exportRecords: ExportRecordsPluginMethod = await getBatchFunction(
-      destination
-    );
+    try {
+      for (const givenExport of givenExports) {
+        if (!givenExport.hasChanges) await givenExport.complete();
```

Comment:
probably just `continue` here. Weird to also cancel it lower or something.
Then can remove the `hasChanges` check below as well.